### PR TITLE
156 157 update dg user login logout

### LINKED ISSUE #156: Update DG - User Login

### LINKED ISSUE #157: Update DG - User Logout

### PR DESCRIPTION
**Pull request by zhonghan721**
Closes #156 
Closes #157 

Update DG to include User Login and Logout implementations

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -193,6 +193,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+The sequence of the `login` command is as follows:
+
+1. Upon launching the application, the `ModelManager` will be initialized with 
+the `User` constructed with details from the authentication.json file.
+2. The user inputs the `login` command with the username and password.
+3. The `userLoginCommandParser` checks whether all the required fields are present.
+If all fields are present, it creates a new `userLoginCommand`.
+4. The `userLoginCommand` checks whether the user is currently logged in by calling `Model#getUserLoginStatus()`. 
+5. The `userLoginCommand` then checks if the user credentials match the stored user by calling `Model#userMatches()`.
```
Comment (B-enguin) [inline docs/DeveloperGuide.md:242]: In the line above it says it check if the user credentials is correct before checking if the user is logged in, but here it is the other way round

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -193,6 +193,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+<puml src="diagrams/UserLoginSequenceDiagram.puml" alt="UserLoginSequenceDiagram" />
+
+### User Logout Command
+
+**Overview:**
+
+The `logout` command is used to log out from the user's account.
+Once logged out, the user will have no access to all the commands available, except for `help`, `exit`, 
+`register` and `delete account`.
```
Comment (B-enguin) [inline docs/DeveloperGuide.md]: Would `login` be considered an available command

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -193,6 +193,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+The format for the `logout` command can be found [here](UserGuide.md#logout).
+
+**Feature details:**
+
+1. The user executes the `logout` command.
+2. If extra command parameters are provided after specifying `logout`, the logout command will still be executed.
+3. If the user is currently logged out, an error message will be shown.
+4. If all the previous steps are completed without exceptions, the user will be logged out and the 
+`isLoggedIn` status in `Model` will be updated to `false`.
+
```
Comment (B-enguin) [inline docs/DeveloperGuide.md:268]: Would it be better if the code corresponding step in the `login` command also indicates that the status is updated to `true`

**Review by B-enguin — COMMENTED** [review]
Some general comments

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (zhonghan721) [reply docs/DeveloperGuide.md:242]: Updated the section prior to this

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (zhonghan721) [reply docs/DeveloperGuide.md:268]: Updated the `login` command section

**Review by jianyangg — APPROVED** [review]
LGTM

**Pull request merged**